ui-library-click-to-call/README.md

```diff
@@ -47,3 +53,1211 @@ Instead, it will copy all the configuration files and the transitive dependencie
+                return (
+                    <img
+                        style={{ height: '4rem', width: '4rem', margin: 'auto' }}
+                        src={hero}
+                        alt="logo"
+                    />
+                );
+            }}
+            onSetDisplayName={setUserDisplayName}
+            onSetUseVideo={setUseVideo}
+        />
+    </Stack>
+    ...
+```
+
+Next we will need to update the `App.tsx` file to listen for and ask for the messages from what would be the parent window. First to start, you might recall that we added a new query parameter to the URL of the application `newSession=true`. To use this and have our app look for that in the URL we will need to create a utility function to parse out that parameter so our application will behave differently when it is received.
+
+To do that lets add a new folder `src/utils` and in this folder we will add the file `AppUtils.ts`. In this file we will put the following function:
+`AppUtils.ts`
+
+```typescript
+/**
+ * get go ahead to request for adapter args from url
+ * @returns
+ */
+export const getStartSessionFromURL = (): boolean | undefined => {
+  const urlParams = new URLSearchParams(window.location.search);
+  return urlParams.get("newSession") === "true";
+};
+```
+
+This function will take a look into our applications URL and see if the param we are looking for is there. You can also stick some other parameters in there
+
+As well we will want to add a new type in here to track the different pieces needed to create a `AzureCommunicationCallAdapter`. This type can also be simplified if you are using our calling stateful client, this approach will not be covered in this tutorial though.
+
+```typescript
+/**
+ * Properties needed to create a call screen for a Azure Communications CallComposite.
+ */
+export type AdapterArgs = {
+  token: string;
+  userId: CommunicationIdentifier;
+  locator: CallAdapterLocator;
+  displayName?: string;
+  alternateCallerId?: string;
+};
+```
+
+Once we have added these two things we can go back to the `App.tsx` file to make some more updates.
+
+First thing we will want to do is update `App.tsx` to use that new utility function that we created above, we will want to use a `useMemo` hook for this so that it is fetched exactly once and not at every render. That is done like so:
+
+`App.tsx`
+```typescript
+// you will need to add these imports
+...
+import { AdapterArgs, getStartSessionFromURL } from './utils/AppUtils';
+...
+```
+
+```typescript
+...
+  const startSession = useMemo(() => {
+    return getStartSessionFromURL();
+  }, []);
+...
+```
+Then, after we have added that we will want to add a event listener like we did earlier to the `ClickToCallScreen.tsx` to the `App.tsx` listening for post messages. Like before we will put it in a `useEffect` hook, this time though we will leave the dependency array empty so that we only add this listener one time on the initial render.
```

Comment:
```suggestion
We now want to add an event listener to `App.tsx` to listen for post messages. Insert a `useEffect` hook with an empty dependency array so that we add the listener only once on the initial render.
```